src/main/java/Deadline.java

```diff
@@ -0,0 +1,17 @@
+
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        // return "[D]" + super.toString() + "(" + by + ")";
+        return "[D]" + (super.isDone ? "[X]" + super.description : "[ ]"+ super.description) + "(" + by + ")" ; // mark done task with X
```

Comment:
Could add new lines to improve readability, and comment could be incorporated as part of javadocs.
Could also clean up line 13

```suggestion
/*
 * @return (what it returns)
 */
               return "[D]" 
                       + (super.isDone ? "[X]" + super.description : "[ ]"+ super.description) 
                       + "(" + by + ")" ; 
```